src/main/kotlin/com/ajax/cryptocurrency/threads/BackgroundJobRunner.kt

```diff
@@ -0,0 +1,24 @@
+package com.ajax.cryptocurrency.threads
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.ApplicationArguments
+import org.springframework.boot.ApplicationRunner
+import org.springframework.stereotype.Component
+
+@Component
+class BackgroundJobRunner (@Autowired private val priceSaver: PriceSaver,
+                           @Value("\${cryptocurrency.name}") private val cryptocurrencyNames: List<String>) : ApplicationRunner {
+    override fun run(args: ApplicationArguments?) {
+        val amountOfThreads = cryptocurrencyNames.size
+        val threads = mutableListOf<Thread>()
+
+        for (i in 1..amountOfThreads) {
+            val thread = Thread(priceSaver)
+            thread.start()
+            threads.add(thread)
+        }
+
+        threads.forEach { it.join() }
```

Comment:
Suggestion to move from java-like style to kotlin-like style:

```kotlin
(1..amountOfThreads)
     .map { Thread(priceSaver) }
     .onEach { it.start() }
     .forEach { it.join() }
```

---- 
Be careful with such approach. You must understand how these extension functions work: map, onEach, forEach.
If you add `.asSequence` before all then all the threads will be run sequentially:

```kotlin
// Wrong approach
(1..amountOfThreads)
     .asSequence()
     .map { Thread(priceSaver) }
     .onEach { it.start() }
     .forEach { it.join() }
```

Reply:
done